include/zephyr/sensing/sensing.h

```diff
@@ -0,0 +1,351 @@
+ * Support multiple Application clients for open same sensor instance,
+ * in this case, the returned handle will different for different clients.
+ * meanwhile, also register sensing callback list.
+ *
+ * @param index The sensor index in sensor info array which get from /ref int sensing_get_sensors().
+ *
+ * @param cb_list callback list to be registered to sensing.
+ *
+ * @param *handle The opened instance handle, if failed will be set to -1.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int sensing_open_sensor(
+		int index, const struct sensing_callback_list *cb_list,
+		int *handle);
+
+/**
+ * @brief Open sensor instance by sensor type and sensor index.
+ *
+ * Application clients use it to open a sensor instance and get its handle.
+ * Support multiple Application clients for open same sensor instance,
+ * in this case, the returned handle will different for different clients.
+ * meanwhile, also register sensing callback list
+ *
+ * @param type The sensor type which need to open.
+ *
+ * @param instance The id for multiple instances with same sensor /ref type, 0 means the default
+ *                 instance for the /ref type.
+ *
+ * @param cb_list callback list to be registered to sensing.
+ *
+ * @param *handle The opened instance handle, if failed will be set to -1.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int sensing_open_sensor_by_type(
+		int type, int instance, const struct sensing_callback_list *cb_list,
+		int *handle);
+
+/**
+ * @brief Close sensor instance.
+ *
+ * @param handle The sensor instance handle need to close.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int sensing_close_sensor(
+		int handle);
+
+/**
+ * @brief Set current config items to Sensor subsystem.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param configs The configs to be set according to config attribute.
+ *
+ * @param count count of configs.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int sensing_set_config(
+		int handle,
+		struct sensing_sensor_config *configs, int count);
+
+/**
+ * @brief Get current config items from Sensor subsystem.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param configs The configs to be get according to config attribute.
+ *
+ * @param count count of configs.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int sensing_get_config(
+		int handle,
+		struct sensing_sensor_config *configs, int count);
+
+/**
+ * @brief Read a data sample.
+ *
+ * This API will trigger target sensor read a sample each time.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer to receive sensor data sample.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int sensing_read_sample(
+		int handle,
+		void *buf, int size);
```

Comment:
Also discarded during the rename. Reading one-shot data should not have its own custom code path. Please use the callbacks to report the data.

Reply:
removed unimplemented API